sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -105,14 +106,13 @@ export class AppConfigurationClient {
    */
   deleteConfigurationSetting(
     key: string,
-    options: DeleteConfigurationSettingOptions & ETagOption
+    options: DeleteConfigurationSettingOptions = {}
   ): Promise<DeleteConfigurationSettingResponse> {
-    if (options.etag) {
-      options = { ...options };
-      options.ifMatch = formatETagForMatchHeaders(options);
-    }
-
-    return this.client.deleteKeyValue(key, options);
+    return this.client.deleteKeyValue(key, {
+      ...options,
+      ifMatch: quoteETag(options.ifMatch),
```

Comment:
Out of curiosity, what happens if both of these are set by the user? Will the service return an error?

Reply:
It'd be nice to come up with a better scheme for representing these conditions because, as you note, this is possible for the user to do.

With getConfigurationSetting the ifMatch condition takes precedence over the the ifNoneMatch condition so it behaves the same as if you'd only passed ifMatch.

Reply:
Decided to do a simple check and it made the rest of the code nicer. So double-bonus!